flink-catalog-aws/flink-catalog-aws-glue/src/main/java/org/apache/flink/table/catalog/glue/TypeMapper.java

```diff
@@ -0,0 +1,124 @@
+            return "double";
+        } else if (logicalType instanceof DateType) {
+            return "date";
+        } else if (logicalType instanceof TimestampType) {
+            return "timestamp";
+        } else if (logicalType instanceof ArrayType) {
+            ArrayType arrayType = (ArrayType) logicalType;
+            String elementType = mapFlinkTypeToGlueType(arrayType.getElementType());
+            return "array<" + elementType + ">";
+        } else if (logicalType instanceof MapType) {
+            MapType mapType = (MapType) logicalType;
+            String keyType = mapFlinkTypeToGlueType(mapType.getKeyType());
+            String valueType = mapFlinkTypeToGlueType(mapType.getValueType());
+            return "map<" + keyType + "," + valueType + ">";
+        } else if (logicalType instanceof RowType) {
+            RowType rowType = (RowType) logicalType;
+            StringBuilder structType = new StringBuilder("struct<");
+            for (RowType.RowField field : rowType.getFields()) {
+                structType
+                        .append(field.getName())
+                        .append(":")
+                        .append(mapFlinkTypeToGlueType(field.getType()))
+                        .append(",");
+            }
+            // Remove the trailing comma and close the struct definition
+            structType.setLength(structType.length() - 1);
+            structType.append(">");
+            return structType.toString();
+        } else {
+            throw new UnsupportedOperationException("Unsupported Flink type: " + logicalType);
+        }
+    }
+
+    public static AbstractDataType<?> glueTypeToFlinkType(String glueType) {
+        switch (glueType) {
+            case "int":
+                return DataTypes.INT();
+            case "bigint":
+                return DataTypes.BIGINT();
+            case "string":
+                return DataTypes.STRING();
+            case "boolean":
+                return DataTypes.BOOLEAN();
+            case "decimal":
+                return DataTypes.DECIMAL(10, 0); // Adjust precision and scale as needed
+            case "float":
+                return DataTypes.FLOAT();
+            case "double":
+                return DataTypes.DOUBLE();
+            case "date":
+                return DataTypes.DATE();
+            case "timestamp":
+                return DataTypes.TIMESTAMP(5);
+            case "array":
+                // Example: array<string> -> DataTypes.ARRAY(DataTypes.STRING())
+                String elementType = glueType.substring(6, glueType.length() - 1);
+                return DataTypes.ARRAY(glueTypeToFlinkType(elementType));
+            case "map":
+                // Example: map<string, string> -> DataTypes.MAP(DataTypes.STRING(),
+                // DataTypes.STRING())
+                int commaIndex = glueType.indexOf(",");
+                String keyType = glueType.substring(4, commaIndex);
```

Comment:
Forgive me for not spotting this earlier - this will always cause an out of bounds exception. The only way this block of code runs is if `glueType` is equal to the string literal `map` but in this block you're attempting to parse the part of the string which cannot exist (`map<string, string>`). It looks like you're expecting `case` here to behave as a `startsWith`. Please address this and consider adding unit tests for this method. I would consider using regex here to parse the key and value types - it will allow you to be more permissive about whitespace (e.g. `map<string, string>` and `map<string,string>`) and also allow you to play with named groups (a personal favourite).